addons/product_expiry/tests/test_stock_lot.py

```diff
@@ -685,3 +685,77 @@ def test_compute_expiration_date_from_scheduled_date(self):
         delivery.action_confirm()
 
         self.assertAlmostEqual(delivery.move_line_ids[0].expiration_date, expiration_date, delta=delta)
+
+    def test_assign_lot_expiry_alert_to_default_user(self):
+        """ Test lot expiry alert is assigned to the default user of the activity type """
+
+        # User A will be the Default User on the activity type
+        user_a, user_b = self.env['res.users'].create([
+            {
+                'name': 'User A (Default)',
+                'login': 'user_a_test',
+            },
+            {
+                'name': 'User B (Responsible)',
+                'login': 'user_b_test',
+            }
+        ])
+
+        activity = self.env.ref('product_expiry.mail_activity_type_alert_date_reached')
+        activity_id = activity.id
+        activity.default_user_id = user_a.id
```

Comment:
This is misleading. your variable `activity` make me think it is a `mail.activity` but it's a `mail.activity.type`